data/api.py

```diff
@@ -67,12 +67,45 @@ def get_queryset(self):
             raise BespinAPIException(400, 'Getting dds-resources requires either a project_id or folder_id query parameter')
 
 
-class WorkflowsViewSet(viewsets.ReadOnlyModelViewSet):
+class ExcludeDeprecatedWorkflowsMixin(object):
+    """
+    Mixin to dynamically build a queryset that excludes deprecated workflows from the listing
+
+    If the exclude() filter is always applied to the queryset, then there's no way to list or access deprecated
+    workflows through the API. They are simply invisible. Instead, we exclude them from the queryset by default, but
+    include them ahead of other filtering or for specific request id.
+    """
+
+    state_filter_field = None
+
+    def get_queryset(self):
+        """
+        Exclude deprecated workflows from queryset, unless request is
+        filtering on state or fetching a single workflow by id
+        :return: Workflows queryset
+        """
+        if self.state_filter_field not in self.filter_fields:
+            raise(BespinAPIException(500, 'Cannot use ExcludeDeprecatedWorkflowsMixin without setting '
+                                          'state_filter_field to one of the filter_fields: {}'
+                                     .format(self.state_filter_field)))
+        queryset = self.queryset
```

Comment:
Is there a reason we are using self.queryset instead of calling super...`get_queryset()`?